src/scripts/domain/moveCalculator.ts

```
@@ -1,53 +1,109 @@
+  private static addPawnMoves(
     board: BoardInfo,
     moves: Move[],
     piece: PieceInfo,
   ) {
     const dY = piece.isWhite ? -1 : 1;
+    const dV = new Vector2(0, dY);
 
     for (let dX = -1; dX <= 1; ++dX) {
-      const move = this.getSimpleMove(board, piece, dX, dY);
+      dV.x = dX;
+      const move = this.getPawnMoveInDirection(board, piece, dV);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)